components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/api/DeviceAuthServiceImpl.java

```diff
@@ -21,25 +21,40 @@
 import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
 import org.wso2.carbon.identity.oauth2.device.constants.Constants;
 import org.wso2.carbon.identity.oauth2.device.dao.DeviceFlowPersistenceFactory;
+import org.wso2.carbon.identity.oauth2.device.model.DeviceFlowDO;
 
 /**
  * Service layer to talk with DAO.
  */
 public class DeviceAuthServiceImpl implements DeviceAuthService {
 
     @Override
+    public String generateDeviceResponse(String deviceCode, String userCode, long quantifier, String clientId,
+                                         String scopes) throws IdentityOAuth2Exception {
+
+        return DeviceFlowPersistenceFactory.getInstance().getDeviceFlowDAO().
+                insertDeviceFlowParametersWithQuantifier(deviceCode, userCode, quantifier, clientId, scopes);
+    }
+
+    @Override
+    @Deprecated
```

Comment:
Point to which method should be used. Also, deprecate from the interface and point to which method should be used instead.